src/libraries/Microsoft.Extensions.DependencyInjection.Specification.Tests/src/ActivatorUtilitiesTests.cs

```diff
@@ -227,12 +229,45 @@ public void TypeActivatorCreateInstanceUsesFirstMathchedConstructor(object value
             var type = typeof(ClassWithAmbiguousCtors);
 
             // Act
-            var instance = ActivatorUtilities.CreateInstance(serviceProvider, type, value);
+            var instance = ActivatorUtilities.CreateInstance(serviceProvider, type, values);
 
             // Assert
             Assert.Equal(ctor, ((ClassWithAmbiguousCtors)instance).CtorUsed);
         }
 
+        [Theory]
+        [InlineData(0, "", "int, IFakeService, string")]
+        public void TypeActivatorCreateInstanceUsesMarkedConstructor(int argument1, string argument2, string ctor)
+        {
+            // Arrange
+            var serviceCollection = new TestServiceCollection();
+            serviceCollection.AddSingleton<IFakeService, FakeService>();
+            var serviceProvider = CreateServiceProvider(serviceCollection);
+            var type = typeof(ClassWithAmbiguousCtorsAndAttributeFirst);
+
+            // Act
+            var instance = ActivatorUtilities.CreateInstance(serviceProvider, type, argument1, argument2);
+
+            // Assert
+            Assert.Equal(ctor, ((ClassWithAmbiguousCtorsAndAttributeFirst)instance).CtorUsed);
+        }
+
+        [Fact]
+        public void TypeActivatorCreateInstanceUsesParameterlessConstructor()
+        {
+            // Arrange
+            var serviceCollection = new TestServiceCollection();
+            serviceCollection.AddSingleton<IFakeService, FakeService>();
+            var serviceProvider = CreateServiceProvider(serviceCollection);
+            var type = typeof(ClassWithParameterlessCtor);
+
+            // Act
+            var instance = ActivatorUtilities.CreateInstance(serviceProvider, type);
+
+            // Assert
+            Assert.Equal("Parameterless", ((ClassWithParameterlessCtor)instance).CtorUsed);
```

Comment:
If IFakeService is in the service collection, it should be using the IFakeService constructor

Reply:
Thing is we only have an `IServiceProvider` at this point and the only way to figure out if a type is registered in DI container is to actually create an instance of that type. Do you feel we should still try check that, despite the possible side effects?